Core/src/org/sleuthkit/autopsy/contentviewers/ContextViewer.java

```diff
@@ -0,0 +1,427 @@
+
+    }//GEN-LAST:event_jSourceGoToResultButtonActionPerformed
+
+    @Override
+    public void setNode(Node selectedNode) {
+        if ((selectedNode == null) || (!isSupported(selectedNode))) {
+            resetComponent();
+            return;
+        }
+
+        AbstractFile file = selectedNode.getLookup().lookup(AbstractFile.class);
+        try {
+            populateSourceContextData(file);
+        } catch (NoCurrentCaseException | TskCoreException ex) {
+            logger.log(Level.SEVERE, "Exception displaying context for file {0}", file); //NON-NLS
+        }
+    }
+
+    @Override
+    public String getTitle() {
+        return Bundle.ContextViewer_title();
+    }
+
+    @Override
+    public String getToolTip() {
+        return Bundle.ContextViewer_toolTip();
+    }
+
+    @Override
+    public DataContentViewer createInstance() {
+        return new ContextViewer();
+    }
+
+    @Override
+    public Component getComponent() {
+        return this;
+    }
+
+    @Override
+    public void resetComponent() {
+        setSourceName("");
+        setSourceText("");
+    }
+
+    @Override
+    public boolean isSupported(Node node) {
+
+        // check if the node has an abstract file and the file has any context defining artifacts.
+        if (node.getLookup().lookup(AbstractFile.class) != null) {
+            AbstractFile abstractFile = node.getLookup().lookup(AbstractFile.class);
+            for (BlackboardArtifact.ARTIFACT_TYPE artifactType : SOURCE_CONTEXT_ARTIFACTS) {
+                List<BlackboardArtifact> artifactsList;
+                try {
+                    artifactsList = abstractFile.getArtifacts(artifactType);
+                    if (!artifactsList.isEmpty()) {
+                        return true;
+                    }
+                } catch (TskCoreException ex) {
+                    logger.log(Level.SEVERE, "Exception while looking up context artifacts for file {0}", abstractFile); //NON-NLS
+                }
+            }
+
+        }
+
+        return false;
+    }
+
+    @Override
+    public int isPreferred(Node node) {
+        return 1;
+    }
+
+    /**
+     * Looks for context providing artifacts for the given file and populates
+     * the source context.
+     *
+     * @param sourceFile File for which to show the context.
+     *
+     * @throws NoCurrentCaseException
+     * @throws TskCoreException
+     */
+    private void populateSourceContextData(AbstractFile sourceFile) throws NoCurrentCaseException, TskCoreException {
+
+        SleuthkitCase tskCase = Case.getCurrentCaseThrows().getSleuthkitCase();
+
+        // Check for all context artifacts
+        boolean foundASource = false;
+        for (BlackboardArtifact.ARTIFACT_TYPE artifactType : SOURCE_CONTEXT_ARTIFACTS) {
+            List<BlackboardArtifact> artifactsList = tskCase.getBlackboardArtifacts(artifactType, sourceFile.getId());
+            if (!artifactsList.isEmpty()) {
+                foundASource = true;
```

Comment:
You could simplify this by just setting foundAsSource = !artifactsList.isEmpty();